### PR TITLE
Guide for Stellar smart contract init templates

### PR DESCRIPTION
**Pull request by CaptainPrinz**
@briwylde08 My code is ready for review!

### CODE REVIEW

**Comment by briwylde08** [comment]
Amazing! We'll get this assigned and reviewed. FYI we have given reviewers three weeks to look things over. 

Solves: https://github.com/stellar/stellar-docs/issues/611

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+### Installing the dependencies
+
+The following dependencies will simplify the process of interacting with files and directories which the `initialize.js` file does together with other functionalities.
+
+1. [glob](https://npmjs.com/package/glob)
+2. [dotenv](https:npmjs.com/package/dotenv)
+
+Install the above packages as dev dependencies by running the code below:
+
+```bash
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:72]:
````suggestion
```bash npm2yarn
````

Just a suggestion, not a "must-have." You can add this to your `bash` code blocks when you are using `npm` or `yarn` commands and it'll provide the options for both, depending on what the user selects.

Comment (ElliotFriend) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:72]: I don't know for sure if it would work in the section above where you're create the project, installing, etc.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+import { globSync } from "glob";
+import path from "path";
+import { writeFileSync, rmSync } from "fs";
+```
+
+### Creating the wrapper function
+
+Instead of calling the `execSync` function and making the error handling process multiple times, we create a wrapper function to encapsulate this function.
+
+Furthermore, when deploying the contract, the command returns the deployed contract ID, which is required when generating contract Typescript bindings. Wrapping the `execSync` function will enable to return Stellar CLI command output while handling errors in one go.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:102]:
```suggestion
Furthermore, when deploying the contract, the command returns the deployed contract address, which is required when generating contract Typescript bindings. Wrapping the `execSync` function will enable the script to return Stellar CLI command output while handling errors in one go.
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+This snippet builds Stellar smart contracts in the project directory.
+
+```javascript
+run("stellar contract build");
+```
+
+### Deploying the contract
+
+This code deploys one or more Stellar smart contracts. It uses files found in target/wasm32-unknown-unknown/release/\*.wasm, deploys each with specified parameters, and stores contract details (alias and contractId) in contractsObject.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:149]:
```suggestion
This code deploys one or more Stellar smart contracts. It uses files found in `target/wasm32-unknown-unknown/release/*.wasm`, deploys each with specified parameters, and stores contract details (alias and contractId) in contractsObject.
```

When writing filepaths or filenames (outside of scripts), let's use backticks to separate them and make it a bit easier for the eye to catch what's happening.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+  contractsObject[alias] = {
+    alias,
+    id: contractId,
+  };
+});
+```
+
+### Generating TypeScript bindings
+
+This snippet generates TypeScript bindings for Stellar smart contracts deployed earlier. It iterates over files and uses `contractsObject` to obtain contract details (alias and `contractId`).
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:173]:
```suggestion
This snippet generates TypeScript bindings for Stellar smart contracts deployed earlier. It iterates over files and uses `contractsObject` to obtain contract details (`alias` and `contractId`).
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+files.forEach((file) => {
+  filename = path.basename(file).split(".")[0];
+  output_dir = `./packages/${filename}`;
+  run(
+    `stellar contract bindings typescript --output-dir ${output_dir} --network ${process.env.NETWORK} --contract-id ${contractsObject[alias]["id"]}  --wasm ${file} --overwrite`,
+  );
+});
+```
+
+### Complete initialize.js script
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:186]:
```suggestion
### Complete initialization script
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+
+The following snippet is the complete aggregated version of the snippets shown earlier. I've added some functionalities like deleting pre-existing contract .wasm files and creating contract.json file, which are all optional steps.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:188]:
```suggestion
The following snippet is the complete aggregated version of the snippets shown earlier. I've added some functionalities like deleting pre-existing contract Wasm files and creating a `contract.json` file, which are all optional steps.
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+  console.error("Error writing JSON file: ", err);
+}
+
+//Generate Contract Typescript Bindings
+let filename, output_dir;
+files.forEach((file) => {
+  filename = path.basename(file).split(".")[0];
+  output_dir = `./packages/${filename}`;
+  run(
+    `stellar contract bindings typescript --output-dir ${output_dir} --network ${process.env.NETWORK} --contract-id ${contractsObject[alias]["id"]}  --wasm ${file} --overwrite`,
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:276]: Is there a reason we're using double-quotes around `["id"]`, but not around `[alias]` when we're referencing the `--contract-id`? Unless there's a reason to prevent it, let's do the same in both, whichever way you want.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+ "workspace": ["./packages/*"],
+```
+
+## Running the script
+
+Now that we have a complete script and a frontend, we should try to run the script. However, because the script is dependent on contract files we need to have a contract project ready, otherwise the script will fail and result in an error. To solve this issue we have to either copy the whole project contents and paste them to an existing Stellar contract project directory or upload the template file to a repo then create a new project and setting the 'f' parameter to point to the hosted repository of the template.
+
+### Setting up the project
+
+I have hosted a [public repository of the React template](https://github.com/CaptainPrinz/soroban-react-template). The following snippet implements the second option:
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:301]:
```suggestion
A community member has hosted a [public repository of the React template](https://github.com/CaptainPrinz/soroban-react-template) available for use. The following command implements the second option:
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+
+```bash
+stellar contract init token_contract -w token -f https://github.com/CaptainPrinz/soroban-react-template.git
+```
+
+The snippet above initializes a new Soroban project, `token_contract` that uses a prebuild token contract and adds the frontend template we deployed to a remote repository.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:307]:
```suggestion
The snippet above initializes a new Stellar smart contracts project, `token_contract`, that uses a prebuilt token contract and adds the frontend template we deployed to a remote repository.
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+
+### Setting the environment variables
+
+Our templates offer a great deal of customization via environment variables. These variables need to be set by the user by creating a `.env` file otherwise the process will fail as these variables will be undefined.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:311]: Let's add a `.env.example` file in the repository, and discuss here how a dev can copy/move that file to `.env` and customize it to fit their needs. You wouldn't need to go into great detail as to the "how," people will figure that out lol. You could also add a link to the example file here somewhere, if you wanted to.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+
+### Starting the script
+
+On your terminal run the following code:
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:315]:
```suggestion
On your terminal run the following command:
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 12]
```diff
@@ -0,0 +1,339 @@
...
+
+```bash
+npm run setup
+```
+
+This runs the `initialize.js` script and first sets the network profile, account identities, builds the contract `.wasm` file, deploys the contracts, and generates the bindings.
+
+If successful, there should be the following changes in the directory:
+
+1. A new `.soroban` directory, which contains the network profile files, account identity files, and hosted smart contract files in json format
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:325]: let's briefly point out how this directory should be added to `.gitignore`

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+2. A `package` directory containing the generated TypeScript bindings
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:326]:
```suggestion
2. A `packages` directory containing the generated TypeScript bindings
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+3. A `contracts.json` file
+
+You are now finally able to import these generated bindings in your frontend application, create their clients, and use them to interact with the contracts deployed on the network.
+
+## Customization
+
+With the `initialize.js` script being the heart of the template file, you can go through the file and modify the paths in which files are saved in if necessary.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:333]:
```suggestion
With the `initialize.js` script being the heart of the template file, you can go through the file and modify the paths in which files are saved if necessary.
```
Redundant `in`.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+
+Also a great deal of customization is enabled by using environment variables, giving flexibility to change network profiles and account identity without limits.
+
+## Conclusion
+
+This section concludes the guide. Check out the documentation on how bindings can be used in dappa as this is a template tutorial is limited to areas concerned with templates as such the processes of connecting these generated bindings and using them in frontend appications is not covered.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:339]:
```suggestion
This section concludes the guide. Check out the documentation on how bindings can be used in dapps as this is a template tutorial is limited to areas concerned with templates as such the processes of connecting these generated bindings and using them in frontend applications is not covered.
```
Couple spelling errors

Comment (ElliotFriend) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:339]: This sentence is also a bit of a run-on. Can you clean up or rephrase this to be a little clearer?

**Review by ElliotFriend — COMMENTED** [review]
Hi, @CaptainPrinz!! This is a really good breakdown of the process and the init script! I've left a few (mostly) minor nitpicks and suggestions throughout.

There's one more thing I think it would be worth touching on. Let's give a simple example of actually using the bindings in the templated project. For example, the [Astro template](https://github.com/stellar/soroban-astro-template/blob/68bd0a88827710525e39014446c6803db03f4f8d/src/pages/index.astro#L5-L6) integrates a `greeting` from the hello world contract into it, so it's not just a blank, empty Astro project after you've initialized it. It's a good touch to make sure that the template actually works "out of the box." This will probably require a little bit of work _in_ the template repo, too. And, you might need to integrate the "Hello World" contract in the `stellar contract init` command, rather than the "Tokens" contract. Then, you can add a brief section like "open up the `path/to/react/component.ts` file and see how we've integrated etc., etc., etc." to this guide.

That change will take this guide from being a useful, descriptive walkthrough of the init script, and turn it into a more full-featured example, as well as a more useable init template overall.

**Comment by CaptainPrinz** [comment]
Okay will go through and make the adjustments

**Comment by CaptainPrinz** [comment]
@ElliotFriend just updated the file with the require section

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 7 above)
Comment (CaptainPrinz) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:277]: alias in this case is a variable whose value we want to use while id is the key we are referencing. putting alias in quotes will break the app

**Comment by CaptainPrinz** [comment]
@ElliotFriend I have implemented most of the suggestions. Waiting for Final checks from you!

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 16]
```diff
@@ -0,0 +1,432 @@
...
+
+If successful, there should be the following changes in the directory:
+
+1. A new `.soroban` directory, which contains the network profile files, account identity files, and hosted smart contract files in json format
+2. A `packages` directory containing the generated TypeScript bindings
+3. A `contracts.json` file
+
+You are now finally able to import these generated bindings in your frontend application, create their clients, and use them to interact with the contracts deployed on the network.
+
+##Interacting with the contract bindings
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:332]:
```suggestion
## Interacting with the contract bindings

```

The heading needs a space after `##`, and an empty line after the heading

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 17]
```diff
@@ -0,0 +1,432 @@
...
+After running the initialization script, there should be new `packages` directory which confirms successful bindings generation. Navigate to the `packages` directory and there should be a package for the deployed contract(s).
+This package will form the last piece in solving the puzzle of interacting with soroban contracts from frontend appications which will be discussed briefly.
+
+To work with the bindings generated, it is recommended the wrap the bindings by creating a new module, instantiating a new client, then finally importing the client and invoking the contract functions.
+
+While it is not necessary to follow the procedure above, you stand to have a better code structure and a much easier way of fixing the code should anything go wrong.
+
+However given the simple structure of our contract, we will proceed to integrate it directly in the frontend without creating a wrapper module.
+
+##Setting up the hello world package
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:342]:
```suggestion
## Setting up the hello world package

```
Same as above

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 18]
```diff
@@ -0,0 +1,432 @@
...
+Navigate to the packages folder to find the hello wold package and observe the directory contents. If the directory does not contain a `dist` folder, you must build the hello world typescript bindings for the package to work in our react app.
+
+```bash
+npm run build
+```
+
+After running the `build` command, there should now be a `dist` directory containing `index.js` and `index.ts` files. These will serve as an entry point to the package when it is eventually imported.
+
+###Using the hello world package in React
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:351]:
```suggestion
### Using the hello world package in React

```
Same as above

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 19]
```diff
@@ -0,0 +1,432 @@
...
+  });
+```
+By printing the `helloWorldContract` object, you'll find the hello method which coincides with the hello function defined in the smart contract. Calling this method will invoke its equivalent in the smart contract.
+
+Finally, to interact with the contract we run the following code:
+```javascript
+ const { result } = await contract.hello({to: receiver});
+```
+
+###Modifying the App component
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:378]:
```suggestion
### Modifying the App component

```
Same as above

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 20]
```diff
@@ -0,0 +1,432 @@
...
+
+## Customization
+
+With the `initialize.js` script being the heart of the template file, you can go through the file and modify the paths in which files are saved if necessary.
+
+Also a great deal of customization is enabled by using environment variables, giving flexibility to change network profiles and account identity without limits.
+
+## Conclusion
+
+Alot has been covered from creating template files to building, deploying, generating bindings, and interacting with smart contracts. Now go ahead and play witht the code and try applying the procedures to other contracts.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:432]:
```suggestion
A lot has been covered from creating template files to building, deploying, generating bindings, and interacting with smart contracts. Now go ahead and play with the code and try applying the procedures to other contracts.
```
Couple spelling nits

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 12 above)
Comment (ElliotFriend) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:326]: Let's **also** add it to your repository's `.gitignore` file, to make sure it's setup for this by default

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 21]
```diff
@@ -0,0 +1,432 @@
...
+## Running the script
+
+Now that we have a complete script and a frontend, we should try to run the script. However, because the script is dependent on contract files we need to have a contract project ready, otherwise the script will fail and result in an error. To solve this issue we have to either copy the whole project contents and paste them to an existing Stellar contract project directory or upload the template file to a repo then create a new project and setting the 'f' parameter to point to the hosted repository of the template.
+
+### Setting up the project
+
+A community member has hosted a [public repository of the React template](https://github.com/CaptainPrinz/soroban-react-template) available for use. The following command implements the second option:
+
+```bash
+stellar contract init `hello_contract` -w hello -f https://github.com/CaptainPrinz/soroban-react-template.git
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:305]:
```suggestion
stellar contract init hello_contract -f https://github.com/CaptainPrinz/soroban-react-template.git
```
The stellar cli (as of `v21.2.0` this morning) will fail if you run this command as-is. The project directory (hello_contract) needs to _not_ be in back-ticks, and `-w hello` is unnecessary, as the hello world contract is included by default. The cli will give an error about it being specified, too.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 22]
```diff
@@ -0,0 +1,432 @@
...
+This package will form the last piece in solving the puzzle of interacting with soroban contracts from frontend appications which will be discussed briefly.
+
+To work with the bindings generated, it is recommended the wrap the bindings by creating a new module, instantiating a new client, then finally importing the client and invoking the contract functions.
+
+While it is not necessary to follow the procedure above, you stand to have a better code structure and a much easier way of fixing the code should anything go wrong.
+
+However given the simple structure of our contract, we will proceed to integrate it directly in the frontend without creating a wrapper module.
+
+##Setting up the hello world package
+Navigate to the packages folder to find the hello wold package and observe the directory contents. If the directory does not contain a `dist` folder, you must build the hello world typescript bindings for the package to work in our react app.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:343]:
```suggestion
Navigate to the packages folder to find the hello world package and observe the directory contents. If the directory does not contain a `dist` folder, you must build the hello world typescript bindings for the package to work in our react app.
```

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 23]
```diff
@@ -0,0 +1,432 @@
...
+### Setting the environment variables
+
+Our templates offer a great deal of customization via environment variables. These variables need to be set by the user by creating a `.env` file otherwise the process will fail as these variables will be undefined.
+
+### Starting the script
+
+On your terminal run the following code:
+
+```bash
+npm run setup
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:319]: Your template repository has a [git conflict saved](https://github.com/CaptainPrinz/soroban-react-template/blob/c2dbaf770005411c87aa2c9b1110aa202cd734fd/initialize.js#L30-L34) in the `initialize.js` file, so this step fails with an error.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 24]
```diff
@@ -0,0 +1,432 @@
...
+However given the simple structure of our contract, we will proceed to integrate it directly in the frontend without creating a wrapper module.
+
+##Setting up the hello world package
+Navigate to the packages folder to find the hello wold package and observe the directory contents. If the directory does not contain a `dist` folder, you must build the hello world typescript bindings for the package to work in our react app.
+
+```bash
+npm run build
+```
+
+After running the `build` command, there should now be a `dist` directory containing `index.js` and `index.ts` files. These will serve as an entry point to the package when it is eventually imported.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:349]: I'm not sure the `build` script actually does what you want it to here. In your template repository, `build` is just calling `vite build`, which only concerns itself with the React app, not the contracts themselves.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 25]
```diff
@@ -0,0 +1,432 @@
...
+        <input type="text" onChange={(e)=> receiver = e.target.value}/>
+        <input type="submit" value="Say hello"/>
+      </form>
+      
+    </>
+  )
+}
+
+export default App
+
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:419]:
```suggestion
```
let's remove the empty line here

File: docs/build/guides/conventions/soroban-contract-init-template.mdx [hunk 26]
```diff
@@ -0,0 +1,432 @@
...
+```
+By printing the `helloWorldContract` object, you'll find the hello method which coincides with the hello function defined in the smart contract. Calling this method will invoke its equivalent in the smart contract.
+
+Finally, to interact with the contract we run the following code:
+```javascript
+ const { result } = await contract.hello({to: receiver});
+```
+
+###Modifying the App component
+To put it all together, modify the `App` component in the `src` directory by replacing its content with the code below.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/soroban-contract-init-template.mdx:379]:
```suggestion
To put it all together, modify the `App.jsx` component in the `src` directory by replacing its content with the code below.
```

**Review by ElliotFriend — COMMENTED** [review]
@CaptainPrinz

You've made some really good progress here! There are still some unresolved issues from the first review that got missed, and I've added a few comments to the newer stuff.

The main thing left to work on is making sure that the template actually works the way you're describing. I went through the process you've outlined here, and there were some errors I ran into along the way. I suggest you walk through the process, starting with the ``stellar contract init `hello_contract` -w hello -f https://github.com/CaptainPrinz/soroban-react-template.git`` command, and fix any trouble (likely in that template repo) along the way.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 24 above)
Comment (CaptainPrinz) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:349]: No In this case I'm referring to the 'build' command in the typescript binding package for the contract. I f you were to observe it's package.json file you'll find the build command

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 21 above)
Comment (CaptainPrinz) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx]: fixed

**Comment by CaptainPrinz** [comment]
I hope this time the issues are fixed once and for all @ElliotFriend .

**Comment by CaptainPrinz** [comment]
@Julian-dev28 @ElliotFriend waiting for more feedback

**Comment by briwylde08** [comment]
Hi @CaptainPrinz! Reviewers will provide further review most likely early next week! Thank you for your patience!

**Comment by briwylde08** [comment]
Hi @CaptainPrinz, sorry for the delay on this one. We'll provide further review asap.

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 24 above)
Comment (ElliotFriend) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx]: Ah! I see. I must have missed the bit about going to the `packages/hello_world` before this command. Thanks!

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 23 above)
Comment (ElliotFriend) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:319]: Running through the process again, two things I've come across:

```suggestion
npm install
cp .env.example .env # edit .env file as needed
npm run setup
```

First, let's be **overt** and say `npm install` needs to run. It's a bit "goes without saying," but I think it'll be common enough for someone going through the tutorial step-by-step to forget about, so let's just be extra clear.

Second, if the `.env.example` file isn't copied over, the initialization script will error out and complain about network and account configs missing.

Third, I'm getting a couple errors when running `npm run setup`. Here's all the output:

```bash
❯ npm run setup          

> hello_contract@0.0.0 setup
> node initialize.js

Configuring Network
Error executing command: 
 Command failed: stellar network add testnet --rpc-url http://soroban-testnet.stellar.org:443 --network-passphrase Test SDF Network ; September 2015
error: unexpected argument 'SDF' found

Usage: stellar network add [OPTIONS] --rpc-url <RPC_URL> --network-passphrase <NETWORK_PASSPHRASE> <NAME>

For more information, try '--help'.
/bin/sh: September: command not found

Here comes the output

Existing Files 
Here comes the output

[ 'target/wasm32-unknown-unknown/release/hello_world.wasm' ]
Deploying Contract hello_world
Here comes the output

JSON file has been saved to ./contracts.json
Error executing command: 
 Command failed: stellar contract bindings typescript --output-dir ./packages/hello_world --network testnet --contract-id CC2L3OKAGKIQZOGP57I76Z7PPVJWTYQ3S6D7SXVQTAMN63TBAFVITWFC
  --wasm target/wasm32-unknown-unknown/release/hello_world.wasm --overwrite
error: --output-dir already exists and you did not specify --overwrite: "./packages/hello_world"
/bin/sh: line 1: --wasm: command not found
```

In your `initialize.js` script, you may want to change [this part](https://github.com/CaptainPrinz/soroban-react-template/blob/2d642ed1aa7e6ad941dead572c1d5a0c43a462eb/initialize.js#L24) to something like this, so the network passphrase gets completely enclosed in quotes.

```js
run(`stellar network add ${process.env.NETWORK} --rpc-url ${process.env.RPC_URL} --network-passphrase "${process.env.NETWORK_PASSPHRASE}"`)
```

There seems to be a trail of other errors in the initialization script that need to be tracked down and figured out. Let me know if you need any help with it.

**Review by ElliotFriend — COMMENTED** [review]
Hi, there @CaptainPrinz! Sorry for the delay.

This is looking really close! I'm having some trouble with the `initialize.js` script in your templates repo. Once following the step-by-step gives us a read-to-go project boilerplate. This'll be ready to merge! 😁 

Also, looks like you've got some MDX formatting problems in the file. If you run `yarn format:mdx` in the `stellar-docs` repository, it should take care of it for you.

**Comment by CaptainPrinz** [comment]
Ill be working on the issues ASAP. As for the reason for the error associated with the scripts, Id have to device a way to get the current OS and write each parameter after a line break. For example, linux uses '/' for line break in terminal like wise powershell has its own, and command prompt has its own as well once its integrated into the code, it'll work

**Comment by CaptainPrinz** [comment]
Apologies @ElliotFriend . You're right putting the network passphrase solves the issue. However I cant seem to find the solution for the script that generates the contracts bindings. I'd appreciate if you could lend me a hand. Ill try as much as possible to crack this. Meanwhile I'm having a really bad internet so this could take a day or two to sort out completely as the upload/download speeds are terrible

File: docs/build/guides/conventions/soroban-contract-init-template.mdx (hunk 23 above)
Comment (CaptainPrinz) [reply docs/build/guides/conventions/soroban-contract-init-template.mdx:323]: Fixed

**Comment by CaptainPrinz** [comment]
Fixed

**Review by ElliotFriend — APPROVED** [review]
Hey, there @CaptainPrinz!!! Thanks **SO MUCH** for all the work you've put into this! It's turned out great, and I'm excited to get it merged in! I figured out a couple of the tiny errors i was getting from the init script, and made CaptainPrinz/soroban-react-template#1 to address them.

Thanks again for getting this across the finish line. Sorry for the busy-ness that led to some delays for the reviews. Well done!

**Comment by CaptainPrinz** [comment]
Glad this worked out. 
Please don't say that's we all have one or more busy-ness to take care of ;)

Once again, I appreciate your effort going through the code

**Comment by briwylde08** [comment]
Hi @CaptainPrinz @ElliotFriend! Just want to confirm this one is ready to merge?

**Comment by ElliotFriend** [comment]
Yeah, let's merge it! 😁

**Comment by briwylde08** [comment]
Amazing! Ok, I'll merge and then we can take the award conversation back to the corresponding issue!
